frontends/concrete-python/concrete/fhe/compilation/utils.py

```diff
@@ -6,20 +6,65 @@
 import os
 import re
 from copy import deepcopy
-from typing import Dict, Iterable, List, Optional, Set, Tuple, Union
+from typing import Any, Callable, Dict, Iterable, List, Optional, Set, Tuple, Union
 
 import networkx as nx
 import numpy as np
 
 from ..dtypes import Float, Integer, SignedInteger, UnsignedInteger
 from ..representation import Graph, Node, Operation
+from ..tracing import ScalarAnnotation
 from ..values import ValueDescription
 from .artifacts import DebugArtifacts
 from .specs import ClientSpecs
 
 # ruff: noqa: ERA001
 
 
+def inputset(
+    *inputs: Union[ScalarAnnotation, ValueDescription, Callable[[int], Any]],
+    size: int = 100,
+) -> List[Tuple[Any, ...]]:
+    """
+    Generate a random inputset.
+
+    Args:
+        *inputs (Union[ScalarAnnotation, ValueDescription, Callable[[int], Any]]):
+            specification of each input
+
+        size (int, default = 100):
+            size of the inputset
+
+    Returns:
+        List[Tuple[Any, ...]]:
+            generated inputset
+    """
+
+    result: List[Tuple[Any, ...]] = []
```

Comment:
can we always add the extrema,
like all input at minima and all input at maxima , or a mix of extrema  ?
how could we ensure the computed bounds are exact ?  (another subject) we should think of bound propagation.

Reply:
> can we always add the extrema,
I don't think it's a good idea as it would make some circuits immediately not compilable. Also it's supposed to be randomly generated inputset.

> how could we ensure the computed bounds are exact ? (another subject) we should think of bound propagation.
We can't think for the ML models, they have `100x100` matmul on 6b for example, theoretically, it can go up to `63 * 63 * 100 = 396900` but it doesn't in practice. We can't work on absolute worst cases because then it just wouldn't compile at all. That's why there is an inputset, it's supposed to represent actual inputs. And this helper is just for creating a random one, it doesn't have worst or best case guarantees.

Reply:
Furthermore, maximum and minimum inputs doesn't always guarantee proper borunds:
```
def f(x):
    assert x.shape == (100,)
    return np.sum(fhe.univariate(lambda x: np.where(x == 42, 1000, 0))(x))
``` 
You can, theoretically, get `100_000` in runtime, how do you propose we find it? This is not an easy problem, we can have very complex logic in subgraphs. And even if we successfully detect all absolute worst bounds, it still won't be useful at all for ML for most applications.